### PR TITLE
I've made your awesome calendar compatible with Godot 3.0.1

### PR DESCRIPTION
**Pull request by Silenoid**
I needed a calendar, found your calendar that was so cool I spent time reworking it a bit to use it in the latest Godot version.

- A couple of changes are just structural, to match Godot new class standards

- I chaged partially the algorythm because of deprecated math expression. I found [an article that talked about the algorythm you used in your work](https://www.hackerearth.com/blog/algorithms/how-to-find-the-day-of-a-week/) and I think I didn't make mistake following that logic.
I think it'd be better to show the article link on your page in case someone needs it

- I've modified the .png button images deleting the sRGB color space that caused some negligible errors

- I've deleted the check_position function just because I didn't need it and wanted to avoid error caused by it. It's easy to correct its behaviour (I've written it in the updated script section)

Thank you so much for what you have done with the calendar. It's been a pleasure working on it :)

### CODE REVIEW

**Comment by ivanskodje** [comment]
Sorry for not noticing this sooner. I will take a look at it when I am back home. :+1:

**Comment by Silenoid** [comment]
Thanks, it's been a pleasure.
Remember that I've deleted the whole position checking procedure just because I'm lazy, so it should be re-implemented using the new margin position variables.
Also, I'm a little worried as the 3.1 update is going to release and predict (and hope) other things will change and add to the engine.

File: addons/calendar_button/calendar_script.gd [hunk 1]
```diff
@@ -147,25 +147,6 @@ func refresh_data():
 			pass
 
 
-# Makes sure the popup window does not leave the screen
```
Comment (ivanskodje) [inline addons/calendar_button/calendar_script.gd:150]: This should be re-implemented in order to provide a GUI-flexible solution. Developers using GUI items such as these, should not have to worry about the backend functionality of the button, and whether or not the popup will appear outside of the window frame

**Review by ivanskodje — CHANGES_REQUESTED** [review]
Thanks for the effort! 

If you can do this without removing the functionality specified in the review, this would be fine. 

Althought from no fault of your own, in Godot 3 the text above the days do not seem to have the correct width. A gold medal bonus if you think you can fix that :1st_place_medal:   (but just re-implementing what you cut away will suffice)

image

**Comment by Silenoid** [comment]
@ivanskodje for the days of the week I've modified the GUI container manually in order to let it position them correctly as intended.
I'm very sorry if I'm dumb and messing with the project :(
When I can, I'll rewrite the position checking of the popup and add the line of code for adjusting the days disposition

**Comment by ivanskodje** [comment]
> @ivanskodje for the days of the week I've modified the GUI container manually in order to let it position them correctly as intended.
> I'm very sorry if I'm dumb and messing with the project :(
> When I can, I'll rewrite the position checking of the popup and add the line of code for adjusting the days disposition

:+1:

**Comment by ccmorataya** [comment]
How about with this? It will be ready for >3.1

**Comment by jospic** [comment]
Hi, 
I've forked original plugin version by Ivan and I've made some improvments as described here:
https://github.com/jospic/godot-plugin-calendar-button

Let me know what you think.
Regards.
-j

File: addons/calendar_button/class/Calendar.gd [hunk 2]
```diff
@@ -68,7 +68,7 @@ func get_number_of_days(m, y):
 # Returns the weekday (int)
 func get_weekday(d, m, y):
 	var t = [0, 3, 2, 5, 0, 3, 5, 1, 4, 6, 2, 4]
-	y -= m < 3
+	if m < 3: y -= m
```
Comment (pattlebass) [inline addons/calendar_button/class/Calendar.gd:71]: Hi, 
Sorry for reviving a dead thread but I might have found a problem with this line.
Shouldn't it be: `if m < 3: y -= 1`

Comment (jospic) [reply addons/calendar_button/class/Calendar.gd:71]: You are right! I had already discovered this bug and fixed it, but I didn't have committed the change yet.
Thanks
-j

Comment (jospic) [reply addons/calendar_button/class/Calendar.gd:71]: just updated my fork repo!

Comment (ivanskodje) [reply addons/calendar_button/class/Calendar.gd:71]: I am very sorry, but could you merge your latest changes with master and reattempt the PR?

I have been away for quite some time, so I have failed to follow-up on this PR. 
The latest version should work as expected, if you made any unique changes/feature improvements, please highlight them :)

**Comment by ivanskodje** [comment]
In hindsight, please go ahead and make a new PR if you are still able / willing to help out 👍  
This will make review easier later on, since a lot in Godot have changed since this was made

**Pull request closed without merging**